pulp_2to3_migration/app/plugin/__init__.py

```diff
@@ -1,8 +1,22 @@
+from gettext import gettext as _
+from collections import namedtuple
+import logging
 import pkg_resources
 
+_logger = logging.getLogger(__name__)
+
 # { plugin_name: PluginMigratorClass }
 PLUGIN_MIGRATORS = {}
+MissingMigrator = namedtuple('MissingMigrator', 'pulp3_plugin')
 
 if not PLUGIN_MIGRATORS:
     for entry_point in pkg_resources.iter_entry_points(group='migrators'):
-        PLUGIN_MIGRATORS[entry_point.name] = entry_point.load()
+        try:
+            PLUGIN_MIGRATORS[entry_point.name] = entry_point.load()
+        except ModuleNotFoundError as exc:
+            _logger.info(_(
+                'Plugin %s is not installed in pulp3 '
+                'therefore it will not be migrated from pulp2') % exc.name
+            )
+            missing = MissingMigrator(exc.msg)
+            PLUGIN_MIGRATORS[entry_point.name] = missing
```

Comment:
@fabricio-aguiar Why do we record which migrators are missed? Where is it used?

@dralley , I think the MP parser relies on the fact that only supported plugins/migrators are present among PLUGIN_MIGRATORS, correct?

Reply:
I recorded due to it:
https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/app/serializers.py#L93-L96
for having a clear message, without recording it, it was falling on:
https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/app/serializers.py#L89-L92

Reply:
I believe so: https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/app/models/base.py#L206

Reply:
to recap the irc discussion:
it seems like https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/app/models/base.py#L206 won't be reached if a migrator is not there, and the presence of not installed plugins allows to fail with a specific message at the MP validation time.